src/schema/python/schemaMigMgr.py

```diff
@@ -67,45 +93,59 @@ def cast(other):
         return Version(other)
 
     @staticmethod
-    def _otherVal(other):
+    def _validateOther(other: object) -> bool:
+        return isinstance(other, (int, Version)) or other is Uninitialized
+
+    @staticmethod
+    def _otherVal(other: object) -> Union[int, Type[Uninitialized]]:
         """Get the value of the other value. May be a ``Version`` instance,
         ``Uninitialized``, or an ``int``.
         """
-        if type(other) is int or other is Uninitialized:
-            return other
-        if type(other) is Version:
+        if not Version._validateOther(other):
+            return NotImplemented
+        if isinstance(other, Version):
             return other.value
-        raise TypeError(f"Unsupported value {other}.")
+        if isinstance(other, int):
+            return int(other)
+        if other is Uninitialized:
+            return Uninitialized
+        raise NotImplementedError(f"_otherVal is not implemented for the type of {other}")
 
-    def __lt__(self, other):
+    def __lt__(self, other: object) -> bool:
         otherVal = self._otherVal(other)
```

Comment:
Do you want to add

        if otherVal  is NotImplemented:
            return NotImplemented

here similarly to other methods?

Reply:
did you mean add these lines like in other methods?
```
if not self._validateOther(other):
    return NotImplemented```

Reply:
`self._otherVal(other)` already calls `_validateOther` and returns `NotImplemented`, so you can use the result of that call. I would probably remove `_validateOther` call from `_otherVal()` and call it explicitly here, because in other places you call `_validateOther` first and `_otherVal` after that, which repeats call to `_validateOther`.

Reply:
good call. thx.